### DISCUSSION TITLE
Ideas / Suggestions

### DISCUSSION THREAD

**Discussion by NateTG** [nicolai-wachenschwan/arc-overhang-prusaslicer-integration#8] [Ideas]
I'm not sure that this is the right place, but there are a couple of thoughts:

I'm not sure how complex it makes things or how well it works in practice, but you may be able to get better results by adding some corrugation (up and down z-movement) to the bridge layer.   This is because making the bridge layer thicker will also make it stiffer, and, without using the z-movement, the thickness is limited to the extrusion width.  

> ... The surface-finish seems to be better with using as little as possible start points for the arcs. but where are the Limits? Finding an algorithm deciding when to start a new arc, working reliable an a wide set of geometries is the next ongoing development.

It seems like the arcs don't have to be concentric at all.  (If they did, you wouldn't be able to paste smaller circles to the sides of big ones like the "recursive" bridging does.)  So, as long as the paths are convex enough, you should be able to make bridges using a technique like this without any "start points."

**Comment by nicolai-wachenschwan** [comment]
Hey NateTG, 
very interesting Idea using additional Z printing... I will move this thought in my head a little and see if any practicable pattern can be derived from this. Maybe spline/bezier curves might be a good candidate...
Currently I dont have time to conduct a search of the minimum convexity, but if you have I would love to hear about it.

Getting rid of the start point would be very good in my opinion bc it is messy to find...
When thinking about patterns please keep in mind that the lower surface quality seems to be very important the the folks... if you have any Idea you could also post a sketch :)

**Reply by NateTG** [reply]
I tried to respond to this earlier and it didn't seem to take.    On reflection, the thing I was thinking of would involve varying the width of the extrusion, and I'm not sure how well the printers support that.

On the math side, "minimum concavity" stuff should be a pretty easy calculation (https://en.wikipedia.org/wiki/Radius_of_curvature).  The maximum radius that's printable is probably going to depend on lots of variables like the type of printer, nozzle size, and print material.

I'll see about making some images of the stuff I'm thinking about later today.

**Reply by NateTG** [reply]
So these arcs stack because they're narrower at their ends than their middles.  The thickness is proportional to the cosine of the angle off center.  Since these go from -45 to +45 degrees, that's a ratio around .707 between the ends and the middle.

![stackingarcs](https://user-images.githubusercontent.com/4387521/223571064-32ece66d-7610-4fdb-babf-2dbfb5ca7d67.png)

**Reply by nicolai-wachenschwan** [reply]
could be worth a shot. unfortunately I dont have any more time. But you can try this out by simply modifying the arc2gcode function, using the length of the arc as reference.
If you want to generate an irregular shape, you can modify the create_circle function as you wish.
